### PR TITLE
Upstream/next release

### PR DESCRIPTION
**Pull request by aroraharsh23**

### CODE REVIEW

File: deploy/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   This topic provides information on how to deploy Citrix node controller on Kubernetes and establish the route between Citrix ADC and Kubernetes Nodes.
 
-**NOTE:** As part of configuration, some resources will be created in the "kube-system" namespace. Hence, Please make sure that "kube-system" namespace is configurable.
+Note: CNC creates "kube-cnc-router" in HOST mode on all the schedulable-nodes. These router pods creates interface on nodes and do iptable config, hence, NET_ADMIN priviledge is required for same. Thus, CNC serviceaccount must have NET_ADMIN priviledge and ability to create HOST mode pods.
```
Comment (dheerajng) [inline deploy/README.md:5]: These router pods create virtual network interface and program iptables accordingly on respective nodes where they are scheduled. These pods need to run with NET_ADMIN capability to achieve the same. Hence CNC serviceaccount must have NET_ADMIN privilege and ability to create HOST mode pods.

File: deploy/README.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ Perform the following:
...
     | VXLAN_PORT | Mandatory | The VXLAN port that you want to use for the overlay. </br></br>**Note:** Ensure that the VXLAN PORT that you use does not conflict with the Kubernetes cluster or Citrix ADC VXLAN PORT. You can use the `show vxlan` command on your Citrix ADC to view the VXLAN PORT. For example: </br></br> `show vxlan` </br>`1) ID: 500       Port: 9090`</br>`Done` </br> </br>In this case, ensure that you do not use `9090` as the VXLAN PORT.|
     | REMOTE_VTEPIP | Mandatory | The Ingress Citrix ADC SNIP. This IP address is used to establish an overlay network between the Kubernetes clusters.|
+    | CNI_TYPE | Mandatory | The CNI used in k8s cluster. Valid values: flannel,calico,canal,weave,cilium|
```
Comment (dheerajng) [inline deploy/README.md:37]: write complete Kubernetes instead of k8s

File: deploy/troubleshoot.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+# Troubleshoot Guide
+
+Quick help Guide for some of the common issues that may arise.
```
Comment (dheerajng) [inline deploy/troubleshoot.md:3]: guide in lowercase.

File: deploy/troubleshoot.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+To validate ADC and basic Node configurations, Refer the screenshots on [deployment](README.md) page.
```
Comment (dheerajng) [inline deploy/troubleshoot.md:5]: 1. Citrix ADC
2. node in lowercase
3. refer in lowercase
4. remove 'the' before screenshots

File: deploy/troubleshoot.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+We can verify few things to debug the issue of services being in DOWN state.
+
+1. Verify logs of CNC pod using:
+
+   ```
+   kubectl logs <cnc-pod> -n <namespace>
+   ```
+
+   Look for any "permission" ERRORs in logs. As stated in deployment page, CNC creates "kube-cnc-router" pods    which need NET_ADMIN priviledge to do the configurations on nodes. So, CNC serviceaccount must have 
+   NET_ADMIN priviledge and the ability to create HOST mode "kube-cnc-router" pods.
```
Comment (dheerajng) [inline deploy/troubleshoot.md:18]: correct spelling of "privilege"

File: deploy/troubleshoot.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+We can verify few things to debug the issue of services being in DOWN state.
+
+1. Verify logs of CNC pod using:
+
+   ```
+   kubectl logs <cnc-pod> -n <namespace>
+   ```
+
+   Look for any "permission" ERRORs in logs. As stated in deployment page, CNC creates "kube-cnc-router" pods    which need NET_ADMIN priviledge to do the configurations on nodes. So, CNC serviceaccount must have
```
Comment (dheerajng) [inline deploy/troubleshoot.md:17]: hyperlink for deployment page is needed here.

File: deploy/troubleshoot.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+   - iptable rule port should created.
+       - port should be same as that of vxlan created on ADC
+       
+
+   <img src="../images/worker-1.png" width="600" height="300">
+
+
+### Service status UP but ping from ADC not working
+
+This is the case wherein though services are UP, still user can't do ping from ADC to the service IP.
```
Comment (dheerajng) [inline deploy/troubleshoot.md:52]: It is possible that user is unable to ping service IP from Citrix ADC even though services are in UP state.

File: deploy/troubleshoot.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+       
+
+   <img src="../images/worker-1.png" width="600" height="300">
+
+
+### Service status UP but ping from ADC not working
+
+This is the case wherein though services are UP, still user can't do ping from ADC to the service IP. 
+One probable reason for this could be the presence of a PBR entry which directs the packets from ADC with SRCIP as NSIP to a default gateway.
+As functionally wise, it will not cause any issue, user if required can ping with SRCIP as ADC VTEP created by CNC. User can use "-s" option to set the SRCIP to SNIP added by CNC on ADC.
```
Comment (dheerajng) [inline deploy/troubleshoot.md:54]: It doesn't impact any functionality. User can ping with Citrix ADC's VTEP as source IP using `-S` option of ping command from Citrix ADC CLI prompt as seen below.
```ping <serviceIP> -S <vtepIP>```

File: deploy/troubleshoot.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+Note: If it is absolutely required to ping with NSIP itself, then as of now, User need to remove the PBR entry or add new PBR entry for endpoint with higher priority
```
Comment (dheerajng) [inline deploy/troubleshoot.md:56]: lowercase of "user"

**Review by dheerajng — CHANGES_REQUESTED** [review]

**Review by subashd — APPROVED** [review]

**Pull request merged**